### DISCUSSION TITLE
Vue example uses loadLocaleAsync in main.ts, but my user language is a user database field

### DISCUSSION THREAD

**Discussion by notflip** [ivanhofer/typesafe-i18n#678] [Q&A]
In an example in the docs I saw this
```
loadLocaleAsync('nl').then(() => {
    app.use(i18nPlugin, 'nl')
    app.mount('#app')
})
```

In my case the user language comes in the JWT token, from my server (it's metadata in the token payload).
I succeeded in getting the correct language in App.vue, as follows

```
onMounted(async () => {
  if (await Session.doesSessionExist()) {
    metadata.value = await Session.getAccessTokenPayloadSecurely()
  }

  await loadLocaleAsync(metadata.value.language)
  setLocale(metadata.value.language)
})
```

But this causes a slight twitch, since 'nl' is loaded in main.ts, and only after the session is available, does the language update.
Any idea how I can tackle this? Thanks

**Comment by ivanhofer** [comment]
This is an issue not directly related to `typesafe-i18n`. Please ask in the vue support channels how you can fix this.
The easiest thing would probably be to wrap your application into an `if` statement, wait for the language to be present and then render the application root.
